developer/src/kmcmplib/src/Compiler.cpp

```diff
@@ -1300,8 +1300,13 @@ KMX_DWORD CheckStatementOffsets(PFILE_KEYBOARD fk, PFILE_GROUP gp, PKMX_WCHAR co
 
         int anyStore = *(q + 2) - 1;
 
-        if (xstrlen(fk->dpStoreArray[indexStore].dpString) < xstrlen(fk->dpStoreArray[anyStore].dpString)) {
-          AddWarning(CWARN_IndexStoreShort); //TODO: if this fails, then we return FALSE instead of an error
+        const int anyLength = xstrlen(fk->dpStoreArray[anyStore].dpString);
+        const int indexLength = xstrlen(fk->dpStoreArray[indexStore].dpString);
+
+        if (indexLength < anyLength) {
+          AddWarning(CWARN_IndexStoreShort);
```

Comment:
So no more TODO about returning FALSE?

Reply:
No, this function returns DWORD, so the TODO is not relevant. (AddWarning is a macro returns a DWORD CERR_xxxx if it fails to log the warning; AddWarningBool is the corresponding function that returns FALSE. Ugly but it is what we have...)